### PR TITLE
Introspect guard

### PR DESCRIPTION
**Pull request by robbytaylor**
FYI @willtj, this adds the ability login a user via introspection.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "designmynight/laravel-oauth-introspect-middleware",
+    "name": "designmynight/laravel-oauth-introspect-client",
     "description": "Laravel Middleware for letting a resource owner verify a OAuth2 access tokens with a remote authorization server",
```
Comment (StevePorter92) [inline composer.json]: Worth updating the description too?

File: src/Introspect.php [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        }
+
+        return $this->result;
+    }
+
+    public function mustHaveScopes(array $requiredScopes = [])
+    {
+        $result = $this->getIntrospectionResult();
+        $givenScopes = explode(' ', $result['scope']);
+        $misingScopes = array_diff($requiredScopes, $givenScopes);
```
Comment (StevePorter92) [inline src/Introspect.php]: typo in `$missingScopes`

**Review by StevePorter92 — APPROVED** [review]
LGTM!

File: src/Introspect.php (hunk 2 above)
Comment (robbytaylor) [reply src/Introspect.php]: Nicely spotted

**Pull request merged**